src/main/java/com/digitalasset/testing/ledger/DamlScriptRunner.java

```diff
@@ -31,31 +36,25 @@ public static Builder builder() {
   public void run() throws IOException, InterruptedException {
     logger.debug("Executing: {}", String.join(" ", processBuilder.command()));
     script = processBuilder.start();
-    if (script.waitFor() != 0) {
+    if (!scriptRunSuccessfully()) {
       throw new IllegalStateException("Unexpected termination of DAML script.");
     }
     logger.info("DAML Script has run successfully.");
   }
 
-  public void kill() {
-    try {
-      if (script.isAlive()) {
-        script.destroyForcibly().waitFor();
-      }
-    } catch (InterruptedException e) {
-      logger.error("Could not stop DAML script.", e);
-    }
+  private boolean scriptRunSuccessfully() throws InterruptedException {
+    return script.waitFor(4, TimeUnit.MINUTES) && script.exitValue() == 0;
```

Comment:
Excessive.
Using `SandboxManager.waitTimeout` might be better. Needs to be threaded through to here, though